dhis-2/dhis-support/dhis-support-hibernate/src/main/java/org/hisp/dhis/datasource/DatabasePoolUtils.java

```diff
@@ -176,25 +178,19 @@ public ConfigKeyMapper getMapper() {
     }
   }
 
+  private DatabasePoolUtils() {}
+
   public static DataSource createDbPool(PoolConfig config)
       throws PropertyVetoException, SQLException {
     Objects.requireNonNull(config);
 
     DbPoolType dbType = DbPoolType.valueOf(config.dbPoolType.toUpperCase());
+    log.info("Database pool type value is [{}]", dbType);
 
-    if (dbType == DbPoolType.C3P0) {
-      return createC3p0DbPool(config);
-    } else if (dbType == DbPoolType.HIKARI) {
-      return createHikariDbPool(config);
-    }
-
-    String msg =
-        String.format(
-            "Database pool type value is invalid, can not create a database pool! Value='%s'",
-            config.dbPoolType);
-    log.error(msg);
-
-    throw new IllegalArgumentException(msg);
+    return switch (dbType) {
+      case C3P0, UNPOOLED -> createC3p0DataSource(config);
```

Comment:
this invalidates a previously resolved comment.
if `UNPOOLED` is the chosen config we should really call the unpooled method (not have it buried in the C3P0 method). Currently, this is very misleading as we are not creating a C3PO pool.

Reply:
ah ok, i thought they were 3 separate, unrelated setups 👍
it needs to be really clear that they are both C3P0 so (the names don't convey this)
would a user know this from the config setting `unpooled`? (unless the docs explain this well)

Reply:
I thought we were getting rid of the now abandoned C3P0 library? Maybe its better to anticipate this now?